src/test/java/io/jenkins/lib/versionnumber/JavaSpecificationVersionTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2019 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package io.jenkins.lib.versionnumber;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.jvnet.hudson.test.For;
+import static io.jenkins.lib.versionnumber.JavaSpecificationVersion.*;
+
+/**
+ * Tests for {@link JavaSpecificationVersion}.
+ */
+@For(JavaSpecificationVersion.class)
+public class JavaSpecificationVersionTest {
+
+    @Test
+    public void shouldParseValidNumbersCorrectly() {
+        assertSpecEquals(JAVA_6, "1.6");
+        assertSpecEquals(JAVA_7, "1.7");
+        assertSpecEquals(JAVA_8, "1.8");
+        assertSpecEquals(JAVA_9, "9");
+        assertSpecEquals(JAVA_10, "10");
+        assertSpecEquals(JAVA_11, "11");
+    }
+
+    @Test
+    public void shouldParseOldSpecCorrectly() {
+        assertSpecEquals(JAVA_9, "1.9");
+        assertSpecEquals(JAVA_10, "1.10");
+        assertSpecEquals(JAVA_11, "1.11");
+        assertSpecEquals(JAVA_12, "1.12");
+    }
+
+    @Test
+    public void shouldResolveIncorrectSpecs() {
+        assertSpecEquals(JAVA_8, "8");
+        assertSpecEquals(JAVA_7, "7");
+        assertSpecEquals(JAVA_5, "5");
+    }
+
+    @Test
+    public void shouldCompareVersionsProperly() {
+        Assert.assertTrue(JAVA_5.isOlderThan(JAVA_6));
+        Assert.assertTrue(JAVA_6.isOlderThan(JAVA_7));
+        Assert.assertTrue(JAVA_7.isOlderThan(JAVA_8));
+        Assert.assertTrue(JAVA_8.isOlderThan(JAVA_9));
+        Assert.assertTrue(JAVA_8.isNewerThan(JAVA_7));
+        Assert.assertTrue(JAVA_9.isOlderThan(JAVA_10));
+        Assert.assertTrue(JAVA_10.isOlderThan(JAVA_11));
+        Assert.assertTrue(JAVA_10.isNewerThan(JAVA_8));
+    }
+
+    @Test
+    public void shouldRetrieveSpecVersionForTheCurrentJVM() {
+        System.out.println("Current JVM Specification Version: " + JavaSpecificationVersion.forCurrentJVM());
```

Comment:
Shouldn't this test have some automatic check (assert)?

Reply:
It is a smoke test. It will blow up if the retrieval logic is not fine. Hence it is a kind of automatic check